daffodil-test/src/test/resources/org/apache/daffodil/section13/text_number_props/TextNumberProps.tdml

```diff
@@ -3462,10 +3467,10 @@
     <xs:element name="nanInvalid" dfdl:lengthKind="delimited" dfdl:terminator="]" type="xsd:double"
     dfdl:textStandardNaNRep="%#r" />
 
-    <xs:element name="infInvalidType" dfdl:lengthKind="delimited" type="xsd:long" dfdl:textStandardInfinityRep="INFINITY"
+    <xs:element name="infInvalidType" dfdl:lengthKind="delimited" type="xsd:long"
     dfdl:terminator="]" />
 
-    <xs:element name="nanInvalidType" dfdl:lengthKind="delimited" type="xsd:long" dfdl:textStandardNaNRep="NOTANUMBER"
+    <xs:element name="nanInvalidType" dfdl:lengthKind="delimited" type="xsd:long"
```

Comment:
This looks like a test bug. I.e., the test should have been written to use a xs:float or xs:double so that the property would be relevant. The test just looks for a parse error complaining about the NOTANUMBER syntax. But if you remove this property, it's not really anything about nan anymore, and so the element name isn't right, nor is the name of the associated test. It's really just a test that non-numeric data won't parse as a long. 

I feel like this test should have been changed into one that is looking for the new warning about dfdl:textStandardNaNRep being ignored on an xsd:long. That's probably covered elsewhere, so arguably, this test here is just plain not relevant anymore, and given the misnomers, should probably just be deleted.

To what degree have tests now evolved into "poorly named tests" that don't really test the things they were originally testing? Is the quality of the tests declining/being-diluted now? Arguably, introduction of this new uniform detection capability for unused properties obviates quite a few tests that were introducting such properties in some sense to test that they were in fact being ignored. 

Do we have a backlog issue now of re-scrutinizing tests to be sure they're still doing meaningful testing?
This is really just a discussion topic that could be held on dev@ mailing list and isn't really core to this review.

Reply:
This is a good point. Compared to how big our test base is, I was actually surprised with how few unused properties there were. And the vast majority of the these were testing that those properties were ignored. They were just unnecessary properties (e.g. boolean props on a string type). So I dont' think this changes things significantly.

In this particular case, this test was really just testing that textStandardNanRep was ignored when the type was xs:long. I think that a reasonable test, but I agree that it probably makes sense to add the property back and expect a warning. I'll take a second pass and make sure there were any tests that I fundamentally changed that should be checking for a warning.

Reply:
I've gone through the modified tests, and the i think there were only three that were actually testing that a property was ignored. One I had already noticed that tested that textNumberPattern was ignored for inputValueCalc. I had already added a dfdl:warnign for that. THe other two were infInvalidType and nanInvalidType, which tests that textStandardInfinityRep and textStandardNaNRep are ignored for a non-float type.

The vast majority of the property removals were related to dfdl:representation on xs:string and xs:hexBInary, or things like text properties defined for a binary representation, which look like they were just copy paste errors. So I think our tests are are still fine and are testing what they say they are testing.

Reply:
Sounds good.